src/backend/python_backend/tests/test_category_routes.py

```
@@ -19,22 +19,32 @@ def test_get_all_categories_db_error(client, mock_db_manager):
-    assert response.status_code == 503
-    assert response.json() == {"detail": "Database service unavailable."}
+    assert response.status_code == 500
+    # Let's match the structure returned by BaseService.handle_error logic
+    response_json = response.json()
+    assert response_json.get("success") is False
+    assert response_json.get("error_code") == "DATABASE_ERROR"
 
 
 def test_create_category(client, mock_db_manager):
     """Test creating a new category successfully."""
+    from src.core.models import CategoryCreate
```

Flake8 (7.3.0)

[error] 'src.core.models.CategoryCreate' imported but unused

(F401)